File: notepad.thenotes/app/src/main/java/com/gbferking/thenotes/ButtonClick.java
package com.gbferking.thenotes;

import android.app.Activity;
import android.graphics.drawable.GradientDrawable;
import android.view.View;

import java.util.Timer;
import java.util.TimerTask;

public class ButtonClick {

    /*
           Criar efeito de Click em Botão em Views
           Use:
               ButtonClick btnclck = new ButtonClick();
           Atribua o evento à view desjeada:
               btnclick.setClick("LinearLayout, RelativeLayout Desejada" linear_button_save, "Activity em uso" LoginActivity.this);

     */

    private Timer timer = new Timer();
    private TimerTask task;
    private int c = 0;

    private void click(View view, Activity activity){
        c++;
        if (c == 1){
            GradientDrawable bg = new GradientDrawable(GradientDrawable.Orientation.TOP_BOTTOM, new int[]{0xFFD9D55B, 0xFFFFEB3B});
            bg.setCornerRadius(80);
            bg.setStroke(3, 0xFFFFF59D);
            view.setElevation(8);
            view.setBackground(bg);
        }
        task = new TimerTask() {
            @Override
            public void run() {
                activity.runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        GradientDrawable bg = new GradientDrawable(GradientDrawable.Orientation.TOP_BOTTOM, new int[]{0xFFFFEB3B, 0xFFD9D55B});
                        bg.setCornerRadius(80);
                        bg.setStroke(3, 0xFFFFF59D);
                        view.setElevation(2);
                        view.setBackground(bg);
                    }
                });
            };
        };
        timer.schedule(task, 200);
    }

    public void setClick(View v, Activity act){
        click(v,act);
    }
}

File: notepad.thenotes/app/src/main/java/com/gbferking/thenotes/ButtonClick.kt
package com.gbferking.thenotes

import android.app.Activity
import android.graphics.drawable.GradientDrawable
import android.view.View
import java.util.Timer
import java.util.TimerTask

class ButtonClick {
    /*
           Criar efeito de Click em Botão em Views
           Use:
               ButtonClick btnclck = new ButtonClick();
           Atribua o evento à view desjeada:
               btnclick.setClick("LinearLayout, RelativeLayout Desejada" linear_button_save, "Activity em uso" LoginActivity.this);

     */
    private val timer = Timer()
    private var task: TimerTask? = null
    private var c = 0

    private fun click(view: View, activity: Activity) {
        c++
        if (c == 1) {
            val bg = GradientDrawable(
                GradientDrawable.Orientation.TOP_BOTTOM,
                intArrayOf(-0x262aa5, -0x14c5)
            )
            bg.cornerRadius = 80f
            bg.setStroke(3, -0xa63)
            view.elevation = 8f
            view.background = bg
        }
        task = object : TimerTask() {
            override fun run() {
                activity.runOnUiThread {
                    val bg = GradientDrawable(
                        GradientDrawable.Orientation.TOP_BOTTOM,
                        intArrayOf(-0x14c5, -0x262aa5)
                    )
                    bg.cornerRadius = 80f
                    bg.setStroke(3, -0xa63)
                    view.elevation = 2f
                    view.background = bg
                }
            }
        }
        timer.schedule(task, 200)
    }

    fun setClick(v: View, act: Activity) {
        click(v, act)
    }
}
